### PR TITLE
Separate region inference logic from error handling better

### PR DESCRIPTION
**Pull request by mark-i-m**
Split out from #67241 

r? @matthewjasper

### CODE REVIEW

File: src/librustc_mir/borrow_check/region_infer/mod.rs [hunk 1]
```diff
@@ -1594,48 +1583,59 @@ impl<'tcx> RegionInferenceContext<'tcx> {
...
+
+            db.buffer(errors_buffer);
+
+            Some(ErrorReported)
+        }
     }
 
-    fn report_or_propagate_universal_region_error(
+    /// Attempt to propagate a region error (e.g. `'a: 'b`) that is not met to a closure's
+    /// creator. If we cannot, then we report an error to the user.
```
Comment (matthewjasper) [inline src/librustc_mir/borrow_check/region_infer/mod.rs]:
```suggestion
    /// creator. If we cannot, then the caller should report an error to the user.
```

**Review by matthewjasper — COMMENTED** [review]
r=me with comment addressed

**Comment by mark-i-m** [comment]
@matthewjasper Done :)

**Pull request merged**